src/input/keybinds.rs

```diff
@@ -0,0 +1,98 @@
+type ModeKeybinds = HashMap<Key, Action>;
+
+/// Populate the default hashmap of keybinds
+/// @@@khs26 What about an input config file?
+pub fn get_default_keybinds() -> Result<Keybinds, String> {
+    let mut defaults = Keybinds::new();
+
+    for mode in InputMode::iter() {
+        defaults.insert(mode, get_defaults_for_mode(&mode)?);
+    }
+
+    Ok(defaults)
+}
+
+fn get_defaults_for_mode(mode: &InputMode) -> Result<ModeKeybinds, String> {
+    let mut defaults = ModeKeybinds::new();
+
+    match *mode {
+        InputMode::Normal => {
+            // Ctrl+G -> Command Mode
+            defaults.insert(Key::Ctrl('g'), Action::SwitchToMode(InputMode::Command));
+        }
+        command_mode @ InputMode::Command | command_mode @ InputMode::CommandPersistent => {
+            match command_mode {
+                InputMode::Command => {
+                    // Ctrl+G -> Command Mode (Persistent)
+                    defaults.insert(
+                        Key::Ctrl('g'),
+                        Action::SwitchToMode(InputMode::CommandPersistent),
+                    );
+                }
+                InputMode::CommandPersistent => {
+                    // Ctrl+G -> Command Mode (Persistent)
+                    defaults.insert(Key::Ctrl('g'), Action::SwitchToMode(InputMode::Normal));
+                }
+                _ => unreachable!(),
+            }
+            // Esc -> Normal Mode
+            defaults.insert(Key::Esc, Action::SwitchToMode(InputMode::Normal));
+            // Resize commands
+            defaults.insert(Key::Char('j'), Action::Resize(Direction::Down));
+            defaults.insert(Key::Char('k'), Action::Resize(Direction::Up));
+            defaults.insert(Key::Char('h'), Action::Resize(Direction::Left));
+            defaults.insert(Key::Char('l'), Action::Resize(Direction::Right));
+            // Move pane commands
+            defaults.insert(Key::Char('u'), Action::MoveFocus(Direction::Down));
+            defaults.insert(Key::Char('i'), Action::MoveFocus(Direction::Up));
+            defaults.insert(Key::Char('y'), Action::MoveFocus(Direction::Left));
+            defaults.insert(Key::Char('o'), Action::MoveFocus(Direction::Right));
+            // Switch focus
+            // @@@ Currently just tab through panes - use right for this
+            defaults.insert(Key::Char('p'), Action::SwitchFocus(Direction::Right));
+            // Scroll
+            defaults.insert(Key::PageUp, Action::ScrollUp);
+            defaults.insert(Key::PageDown, Action::ScrollDown);
+            // Tab controls
+            defaults.insert(Key::Char('1'), Action::NewTab);
+            defaults.insert(Key::Char('2'), Action::GoToNextTab);
+            defaults.insert(Key::Char('3'), Action::GoToPreviousTab);
+            defaults.insert(Key::Char('4'), Action::CloseTab);
+            // New pane
+            defaults.insert(Key::Char('z'), Action::NewPane(Direction::Down));
```

Comment:
@khs26 - I think the bug in the tests is here. ctrl-z (actually ctrl-m, I think the comment was wrong but I don't remember why it was changed, so I guess either one will do :) ) should not have a direction. This should send `PtyInstruction::NewPane` which should find the largest free space and spawn a pane there (horizontally or vertically, it makes the decision on its own)